### PR TITLE
inBasicDraw with bishops of different color

### PR DESCRIPTION
**Pull request by jmogilner**
Endgame with only bishops of different color should be consider a basic draw too.
- Logic added
- Test added (and removed the opposite case)
- Refactored isBasicDraw to simplify the logic

### CODE REVIEW

File: src/Chess/Game/ChessGame.php [hunk 1]
```diff
@@ -1425,90 +1425,42 @@ public function in50MoveDraw()
...
-                    } elseif (array_key_exists('B', $pieces['W']) && is_array($pieces['W']['B']) && count($pieces['W']['B']) > 1) {
-                        return false;
-                    } else {
-                        return true;
-                    }
-                } else {
-                    return false;
-                }
-            }
+        if ($piecesCount == 2) {
```
Comment (marcosdsanchez) [inline src/Chess/Game/ChessGame.php]: You can do a strict check here.

File: src/Chess/Game/ChessGame.php [hunk 2]
```diff
@@ -1425,90 +1425,42 @@ public function in50MoveDraw()
...
-                    if (array_key_exists('N', $pieces['B']) && is_array($pieces['B']['N']) && count($pieces['B']['N']) > 1) {
-                        return false;
-                    } elseif (array_key_exists('B', $pieces['B']) && is_array($pieces['B']['B']) && count($pieces['B']['B']) > 1) {
-                        return false;
-                    } else {
-                        return true;
-                    }
-                } else {
-                    return false;
+        if ($piecesCount == 4) {
```
Comment (marcosdsanchez) [inline src/Chess/Game/ChessGame.php]: You can do a strict check here.

File: src/Chess/Game/ChessGame.php [hunk 3]
```diff
@@ -1425,90 +1425,42 @@ public function in50MoveDraw()
...
+            $playerWith3Pieces = count($whitePieces) === 3 ? $whitePieces : $blackPieces;
+
+            if (in_array($playerWith3Pieces[0], array('K', 'N'))
+                && in_array($playerWith3Pieces[1], array('K', 'N'))
+                && in_array($playerWith3Pieces[2], array('K', 'N'))) {
+                return true; //KNN vs K
             }
         }
 
+        $playerWith2Pieces = count($whitePieces) == 2 ? $whitePieces : $blackPieces;
```
Comment (marcosdsanchez) [inline src/Chess/Game/ChessGame.php]: Use strict check here.

File: src/Chess/Game/ChessGame.php [hunk 4]
```diff
@@ -3829,4 +3781,35 @@ private function getPieceTypes()
...
+     *
+     * Used to determine basic draw conditions
+     * @return array Format:
+     *
+     * <pre>
+     * array(
+     *   // white pieces
+     *   'W' => array('B', 'B', 'K'), //2 bishops and the knight
+     *   // black pieces
+     *   'W' => array('Q', 'K'), //Queen and the knight
```
Comment (marcosdsanchez) [inline src/Chess/Game/ChessGame.php]: `'B'`

File: src/Chess/Game/ChessGame.php [hunk 5]
```diff
@@ -1425,90 +1425,42 @@ public function in50MoveDraw()
...
                 }
             }
-        }
-        $wpIndex = ($whitePieces[0] == 'K') ? 1 : 0;
-        $bpIndex = ($blackPieces[0] == 'K') ? 1 : 0;
-        if ($whitePieces[$wpIndex] == 'B' && $blackPieces[$bpIndex] == 'B') {
-            // bishops of same color?
-            if ($pieces['B']['B'][0] == $pieces['W']['B'][0]) {
-                return true;
+
```
Comment (marcosdsanchez) [inline src/Chess/Game/ChessGame.php:1448]: Please add a test case to cover this scenario:

2020-01-15-153808_3173x563_scrot

**Review by marcosdsanchez — APPROVED** [review]
Thanks. Please see comments.

File: src/Chess/Game/ChessGame.php (hunk 5 above)
Comment (jmogilner) [reply src/Chess/Game/ChessGame.php:1448]: Added

**Pull request merged**